opal/core/patient_lists.py

```diff
@@ -119,6 +119,69 @@ def get_template_prefixes(self):
         return possible
 
 
+"""
+Sometimes we group lists for display purposes.
+"""
+class TabbedPatientListGroup(discoverable.DiscoverableFeature):
+    """
+    Groups of Patient Lists to display as tabs at the top of
+    any list in the group
+    """
+    module_name   = 'patient_lists'
+    member_lists  = []
+
+    @classmethod
+    def for_list(klass, patient_list):
+        """
+        Returns the group for a given PatientList.
+        Raises ValueError if not passed a PatientList
+        """
+        msg = 'TabbedPatientListGroup.for_list must be passed a PatientList'
+        try:
+            if not issubclass(patient_list, PatientList):
+                raise ValueError(msg)
+        except TypeError:
+                raise ValueError(msg)
+
+        for group in klass.list():
+            if patient_list in group.get_member_lists():
+                return group
+
+    @classmethod
+    def get_member_lists(klass):
+        """
+        A hook for dynamically customising the members of this list group.
+
+        Returns an iterable of PatientLists
+        Defaults to the `.member_lists` property
+        """
+        for l in klass.member_lists:
+            yield l
```

Comment:
should we add user as an optional argument here? Either that or just change klass.member_lists to be a tuple and then just return klass.member_lists